### PR TITLE
[feature](Nereids): remove useless project

### PR DESCRIPTION
**Pull request by jackwener**
## Proposed changes

If a projection select all columns, we can eliminate it

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/nereids/processor/post/EliminateProject.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    @Override
+    public Plan visitPhysicalProject(PhysicalProject<? extends Plan> project, CascadesContext ctx) {
+        Plan child = project.child();
+        Plan newChild = child.accept(this, ctx);
+        if (project.isAllSlots()) {
+            Set<Slot> projects = project.getProjects().stream().map(Slot.class::cast).collect(Collectors.toSet());
+            Set<Slot> outputSet = newChild.getOutputSet();
+            if (outputSet.equals(projects)) {
+                return ((AbstractPhysicalPlan) newChild).copyStatsAndGroupIdFrom(project);
+            }
```
Comment (924060929) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/processor/post/EliminateProject.java:45]: I think this two set should change to list.
The project is useless if and only if the order, output number, and each slots are equals.

e.g. the top project maybe use to change the output order:
```sql
-- the tbl output: id, name
select name, id
from tbl
```

**Comment by jackwener** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/processor/post/EliminateProject.java (hunk 1 above)
Comment (jackwener) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/processor/post/EliminateProject.java:45]: `ResultSink` already confirm the order of output

**Comment by 924060929** [comment]
why not use `EliminateUnnecessaryProject` and change it to support `PhysicalProject`?

**Pull request closed without merging**